src/devices/Bmp280/Bmp280.cs

```diff
@@ -0,0 +1,420 @@
+            var2 = var2 + ((long)_calibrationData.DigP4 << 35);
+            var1 = ((var1 * var1 * (long)_calibrationData.DigP3) >> 8) + ((var1 * (long)_calibrationData.DigP2) << 12);
+            var1 = (((((long)1 << 47) + var1)) * (long)_calibrationData.DigP1) >> 33;
+            if (var1 == 0)
+            {
+                return 0; //Avoid exception caused by division by zero
+            }
+            //Perform calibration operations
+            long p = 1048576 - adcPressure;
+            p = (((p << 31) - var2) * 3125) / var1;
+            var1 = ((long)_calibrationData.DigP9 * (p >> 13) * (p >> 13)) >> 25;
+            var2 = ((long)_calibrationData.DigP8 * p) >> 19;
+            p = ((p + var1 + var2) >> 8) + ((long)_calibrationData.DigP7 << 4);
+            return p;
+        }
+
+        /// <summary>
+        ///  Reads an 8 bit value from a register
+        /// </summary>
+        /// <param name="register">
+        ///  Register to read from
+        /// </param>
+        /// <returns>
+        ///  Value from register
+        /// </returns>
+        private byte Read8BitsFromRegister(byte register)
+        {
+            if (_communicationProtocol == CommunicationProtocol.I2c)
+            {
+                _i2cDevice.WriteByte(register);
+                byte value = _i2cDevice.ReadByte();
+                return value;
+            }
+            else
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        /// <summary>
+        ///  Reads a 16 bit value over I2C 
+        /// </summary>
+        /// <param name="register">
+        ///  Register to read from
+        /// </param>
+        /// <returns>
+        ///  Value from register
+        /// </returns>
+        internal ushort Read16BitsFromRegister(byte register)
+        {
+            if (_communicationProtocol == CommunicationProtocol.I2c)
+            {
+                byte[] readBuffer = { 0x00, 0x00 };
+
+                _i2cDevice.WriteByte(register);
+                _i2cDevice.Read(readBuffer);
+
+                return BinaryPrimitives.ReadUInt16LittleEndian(readBuffer);
+            }
+            else
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        /// <summary>
+        ///  Reads a 24 bit value over I2C 
+        /// </summary>
+        /// <param name="register">
+        ///  Register to read from
+        /// </param>
+        /// <returns>
+        ///  Value from register
+        /// </returns>
+        private uint Read24BitsFromRegister(byte register)
+        {
+            if (_communicationProtocol == CommunicationProtocol.I2c)
+            {
+                byte[] readBuffer = { 0x00, 0x00, 0x00 };
+
+                _i2cDevice.WriteByte(register);
+                _i2cDevice.Read(readBuffer);
+
+                return BinaryPrimitives.ReadUInt32LittleEndian(readBuffer);
```

Comment:
I'm not 100% sure but I believe this will throw because readBuffer doesn't have 4 bytes which is what an Int32 requires. In order to satisfy this, you could have your readBuffer be of size 4, and just have the readBuffer[0] to be 0x00 and start the read from readBuffer[1].

Reply:
Fixed